### PR TITLE
Fix python2 enforcement in patch_sys_executable

### PR DESCRIPTION
**Pull request by jdemaeyer**
`python --version` print version to stderr, not stdout

### CODE REVIEW

File: shub/utils.py [hunk 1]
```diff
@@ -100,7 +100,9 @@ def patch_sys_executable():
             try:
                 # Will raise NotFoundException if no Python installation found
                 py_exe = find_exe('python')
-                if '2.' not in get_cmd_output([py_exe, '--version']):
+                py_ver = subprocess.check_output([py_exe, '--version'],
+                                                 stderr=subprocess.STDOUT)
+                if '2.' not in py_ver:
```
Comment (dangra) [inline shub/utils.py]: I am tempted to be more strict and check for `^Python 2.7.`

Comment (jdemaeyer) [inline shub/utils.py]: Just realized the current test also matches `3.2.0` :see_no_evil:
I'm fine with checking 2.7. I thought Ubuntu Precise was still on 2.6 but even they have upgraded. Why the `^Python`?

Comment (dangra) [inline shub/utils.py]: it was a way to say `py_ver.startswith('Python 2.7.')`

Comment (jdemaeyer) [inline shub/utils.py]: I'm a little afraid of unnecessarily excluding non-CPython implementations. E.g. what if PyPy outputs "PyPy 2.7.11". But then again I guess it doesn't, it is the language of the version and not the interpreter after all.

**Pull request closed without merging**